contracts/dao/ListaRevenueDistributor.sol

```diff
@@ -112,6 +124,58 @@ contract ListaRevenueDistributor is Initializable, AccessControlUpgradeable {
         emit RevenueDistributed(token, amount0, amount1);
     }
 
+    /**
+     * @dev distribute tokens to autoBuybackAddress and revenueWalletAddress
+     *      according to distributeRate excluding cost
+     *
+     * @param tokens, token to distribute
+     * @param costs, amount directly to cost address
+     */
+    function distributeTokensWithCost(address[] memory tokens, uint256[] memory costs)
+        external
+        onlyRole(MANAGER)
+    {
+        for (uint256 i = 0; i < tokens.length; i++) {
+            _distributeTokenWithCost(tokens[i], costs[i]);
+        }
+    }
+
+    function _distributeTokenWithCost(address token, uint256 cost) internal {
+        require(tokenWhitelist[token], "token not whitelisted");
+        require(tokenCostToAddress != address(0), "reserve address not set");
+
+        uint256 balance = IERC20(token).balanceOf(address(this));
+        if (balance == 0) {
+            return;
+        }
+
+        if (balance <= cost) {
+            IERC20(token).safeTransfer(tokenCostToAddress, balance);
+            emit RevenueDistributedWithCost(token, 0, 0, balance, cost);
+        } else {
+            uint256 available = balance - cost;
+            uint256 amount0 = available * distributeRate / RATE_DENOMINATOR;
+            uint256 amount1 = available - amount0;
+            if (amount0 > 0) {
+                if (token == listaTokenAddress) {
+                    // lista should skip autoBuyback process
+                    IERC20(token).safeTransfer(listaDistributeToAddress, amount0);
+                } else {
+                    IERC20(token).safeTransfer(autoBuybackAddress, amount0);
+                }
+
+            }
+            if (amount1 > 0) {
+                IERC20(token).safeTransfer(revenueWalletAddress, amount1);
+            }
+            if (cost > 0) {
+                IERC20(token).safeTransfer(tokenCostToAddress, cost);
```

Comment:
In case of exact balance and using fee on transfer tokens, the remaining will be less than the cost, so this would fail.

Either can add in comment if those tokens are not supported or do a before and after balance check.